From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Xiaoyin Liu <xiaoyinl@users.noreply.github.com>
Date: Tue, 15 Nov 2022 07:25:34 -0500
Subject: [PATCH] 4749: Parser: Remove -DSTRSAFE_LIB and STRSAFE_LIB_IMPL

Remove -DSTRSAFE_LIB and STRSAFE_LIB_IMPL

Remove -DSTRSAFE_LIB and -DSTRSAFE_LIB_IMPL from C Parser
profiles and from GDT parser scripts. If they are defined, functions
like StringCbCopyW() will not be included in the archive.

Remove -DSTRSAFE_LIB from VS22_64.prf
---
 Ghidra/Features/Base/data/parserprofiles/VisualStudio12_32.prf  | 2 --
 Ghidra/Features/Base/data/parserprofiles/VisualStudio12_64.prf  | 2 --
 Ghidra/Features/Base/data/parserprofiles/VisualStudio22_64.prf  | 2 --
 Ghidra/Features/Base/data/parserprofiles/vs12Local.prf          | 2 --
 .../Base/ghidra_scripts/CreateExampleGDTArchiveScript.java      | 2 --
 5 files changed, 10 deletions(-)

diff --git a/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_32.prf b/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_32.prf
index 2a108c0430..642d453bf9 100644
--- a/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_32.prf
+++ b/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_32.prf
@@ -185,8 +185,6 @@ rpcproxy.h
 -D_USE_ATTRIBUTES_FOR_SAL
 -D_CRTBLD
 -D_OPENMP_NOFORCE_MANIFEST
--DSTRSAFE_LIB
--DSTRSAFE_LIB_IMPL
 -DLPSKBINFO=LPARAM
 -DCONST=const
 -D_CRT_SECURE_NO_WARNINGS
diff --git a/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_64.prf b/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_64.prf
index 1793a6c2d4..42163f28d3 100644
--- a/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_64.prf
+++ b/Ghidra/Features/Base/data/parserprofiles/VisualStudio12_64.prf
@@ -187,8 +187,6 @@ rpcproxy.h
 -D_USE_ATTRIBUTES_FOR_SAL
 -D_CRTBLD
 -D_OPENMP_NOFORCE_MANIFEST
--DSTRSAFE_LIB
--DSTRSAFE_LIB_IMPL
 -DLPSKBINFO=LPARAM
 -DCONST=const
 -D_CRT_SECURE_NO_WARNINGS
diff --git a/Ghidra/Features/Base/data/parserprofiles/VisualStudio22_64.prf b/Ghidra/Features/Base/data/parserprofiles/VisualStudio22_64.prf
index 29f8ccfcf2..e715ea7bec 100644
--- a/Ghidra/Features/Base/data/parserprofiles/VisualStudio22_64.prf
+++ b/Ghidra/Features/Base/data/parserprofiles/VisualStudio22_64.prf
@@ -305,8 +305,6 @@ lmwksta.h
 -D_USE_ATTRIBUTES_FOR_SAL
 -D_CRTBLD
 -D_OPENMP_NOFORCE_MANIFEST
--DSTRSAFE_LIB
--DSTRSAFE_LIB_IMPL
 -DLPSKBINFO=LPARAM
 -DCONST=const
 -D_CRT_SECURE_NO_WARNINGS
diff --git a/Ghidra/Features/Base/data/parserprofiles/vs12Local.prf b/Ghidra/Features/Base/data/parserprofiles/vs12Local.prf
index cdf7b4b8bd..dfb034202d 100644
--- a/Ghidra/Features/Base/data/parserprofiles/vs12Local.prf
+++ b/Ghidra/Features/Base/data/parserprofiles/vs12Local.prf
@@ -233,8 +233,6 @@ wspiapi.h
 -DCRTDLL
 -D_CRTBLD
 -D_OPENMP_NOFORCE_MANIFEST
--DSTRSAFE_LIB
--DSTRSAFE_LIB_IMPL
 -DLPSKBINFO=LPARAM
 -DCONST=const
 -D_CRT_SECURE_NO_WARNINGS
diff --git a/Ghidra/Features/Base/ghidra_scripts/CreateExampleGDTArchiveScript.java b/Ghidra/Features/Base/ghidra_scripts/CreateExampleGDTArchiveScript.java
index 7a5d52bcf1..60de9e4c4b 100644
--- a/Ghidra/Features/Base/ghidra_scripts/CreateExampleGDTArchiveScript.java
+++ b/Ghidra/Features/Base/ghidra_scripts/CreateExampleGDTArchiveScript.java
@@ -470,8 +470,6 @@ public class CreateExampleGDTArchiveScript extends GhidraScript {
 				"-D_USE_ATTRIBUTES_FOR_SAL",
 				"-D_CRTBLD",
 				"-D_OPENMP_NOFORCE_MANIFEST",
-				"-DSTRSAFE_LIB",
-				"-DSTRSAFE_LIB_IMPL",
 				"-DLPSKBINFO=LPARAM",
 				"-DCONST=const",
 				"-D_CRT_SECURE_NO_WARNINGS",
-- 
2.45.1

